sdk/eventhub/event-hubs/src/linkEntity.ts

```diff
@@ -124,7 +129,23 @@ export class LinkEntity {
     await defaultLock.acquire(this._context.cbsSession.cbsLock, () => {
       return this._context.cbsSession.init();
     });
-    const tokenObject = await this._context.tokenProvider.getToken(this.audience);
+    let tokenObject: AccessToken;
+    let tokenType: TokenType;
+    if (this._context.tokenCredential instanceof SharedKeyCredential) {
+      tokenObject = this._context.tokenCredential.getToken(this.audience);
+      tokenType = TokenType.CbsTokenTypeSas;
+      // renew sas token in every 45 minutess
+      this._tokenTimeout = (3600 - 900) * 1000;
+    } else {
+      const aadToken = await this._context.tokenCredential.getToken(Constants.aadEventHubsScope);
+      if (!aadToken) {
+        throw new Error("Aad token cannot be null");
+      }
+      tokenObject = aadToken;
+      tokenType = TokenType.CbsTokenTypeJwt;
+      this._tokenTimeout = tokenObject.expiresOnTimestamp - Date.now() - 2 * 60 * 1000;
```

Comment:
Take care of all the corner scenarios like
- expiresOn < Date.now()
- expiresOn + 2minutes < Date.now()

Both will result in `_tokenTimeout` being -ve value. If the setTimeout/setInterval functions handle -ve values well, then not a problem I guess

Reply:
I think we should handle these cases whether or not setTimeout function handles -ve values well.

I'm thinking to set the default `_tokenTimeout`(45 mintutes same as sas token) in these cases? what do you think?

Reply:
-ve values mean that the expiry time has already passed. Using 45 mins at this point doesnt help. I would use 0

Reply:
In that case  setTimeout function treats -ve values as 0, so we don't need to add these checks.